### PR TITLE
Add MacPorts support with preconditions

### LINKED ISSUE #32: Alternatives to HomeBrew
Feature request: I think riddance from Brew would be a good move, many (old-fashioned) devs still don't believe in extending on top of the OS X utils and are using MacPorts instead. For reliability reasons.

### PR DESCRIPTION
**Pull request by ain**
Resolve #32 Alternatives to HomeBrew by MacPorts with preconditions:
- Boot2Docker
- Docker Compose

### CODE REVIEW

**Comment by brikis98** [comment]
Thank you for the pull request. I like the general approach and I think this could be made to work, but as I mentioned in #32, before increasing the complexity of the installer, we need to find a way to add good automated tests, which is currently blocked by #7.

On a side note, this PR seems to have a lot of whitespace changes, which makes it hard to identify the real changes. Could you check your text editor settings for what could be causing that?

File: src/setup.sh [hunk 1]
```diff
@@ -273,7 +288,34 @@ function get_env_file {
...
+#
+function is_port_installed {
+  type port > /dev/null 2>&1
+}
+
+#
+# Checks if all MacPorts requirements are met.
+#
+function check_port_prerequisites {
+  if ! type boot2docker > /dev/null 2>&1 ; then
```
Comment (brikis98) [inline src/setup.sh]: Boot2Docker is one of the items in the `port_install` list. Why is it a prerequisite?

File: src/setup.sh [hunk 2]
```diff
@@ -273,7 +288,34 @@ function get_env_file {
...
+    log_error "MacPorts-based setup expects Docker Compose to be preinstalled."
```
Comment (brikis98) [inline src/setup.sh]: Looks like a copy/paste error. This error message should be about Boot2Docker and the one below should be about Docker Compose.

File: src/setup.sh [hunk 3]
```diff
@@ -273,7 +288,34 @@ function get_env_file {
...
+    exit 1
+  elif ! type docker-compose > /dev/null 2>&1 ; then
```
Comment (brikis98) [inline src/setup.sh]: Is there no MacPorts Docker Compose recipe?

File: src/setup.sh (hunk 1 above)
Comment (ain) [inline src/setup.sh]: One of the reasons why a single install method is a bad idea and would grow
into conditional spaghetti: MacPorts has no Portfile for Boot2Docker or
Docker Compose.

File: src/setup.sh [hunk 4]
```diff
@@ -294,19 +336,32 @@ function check_prerequisites {
...
-  brew_install "boot2docker" "Boot2Docker" "boot2docker" false
-  brew_install "docker-compose" "Docker Compose" "docker-compose" false
-  brew_install "fswatch" "fswatch" "fswatch" false
-  brew_install "coreutils" "GNU core utilities" "greadlink" false
+  elif is_port_installed ; then
+
+    log_info "Updating MacPorts"
+    sudo port selfupdate
+
+    port_install "virtualbox" "VirtualBox" "VBoxManage"
```
Comment (brikis98) [inline src/setup.sh]: I'm a little concerned that most of the HomeBrew list is duplicated here, but not all. It makes it trickier to maintain, as you have to update two places (albeit, at least here they are close together), and it's not obvious why some of the items are in one list but not another. Although bash is terrible at data structures, I'm wondering if there is a way to create one to make the solution look more obvious. For example, in JavaScript, I might have used an object like this:

``` js
var dependencies = [
  {packageName: "virtualBox", readableName: "Virtual Box", commandName: "VBoxManage", brew: false, brewCask: true, ports: true},
  {packageName: "docker-compose", readableName: "Docker Compose", commandName: "docker-compose", brew: true, brewCask: false, ports: false},
  // ...
]
```

Comment (ain) [inline src/setup.sh]: Good call, but you have to consider possibly different naming conventions
as well.

Comment (brikis98) [inline src/setup.sh]: True. Perhaps this format is more like it:

``` js
var dependencies = [
  {readableName: "Virtual Box", commandName: "VBoxManage", brewName: null, brewCaskName: "virtualbox", portsName: "virtualbox"},
  {readableName: "Docker Compose", commandName: "docker-compose", brewName: "docker-compose", brewCaskName: null, portsName: null},
  // ...
]
```

Comment (ain) [inline src/setup.sh]: Yup. If nesting is doable, I'd go Managers > Brew|Ports > PackageName,
inside a package descriptor, so you can add others later, e.g. pip or
whatever.

**Comment by ain** [comment]
Dumping maintenance on this PR. Diverged.

**Pull request closed without merging**